### ISSUE TITLE
Proposal: Create a Way to Find Local Service with DNS Discovery

### ISSUE THREAD

**Issue by ekristen** [hashicorp/consul#1229]
In some deployment models like with `nsq` it is recommended that you run a `nsq` instance per server and all applications on that server talk to that `nsq` instance. 

As it stands today I either have to pass the node name or manually specify the port and host to connect to for all apps on the server.

My proposal is that the service tag `local` is introduced as a reserved tag that when seen by the agent means it looks only locally for that service and returns the appropriate DNS.

Example:

* `local.nsqd.service.consul` on server01 will return 10.10.100.100
* `local.nsqd.service.consul` on server02 will return 10.10.100.200
* `nsqd.service.consul` on server01 will return 10.10.100.100, 10.10.100.200
* `nsqd.service.consul` on server02 will return 10.10.100.100, 10.10.100.200

To elaborate a little further, the `local` tag should work like normal, only return DNS record if the service exists on the local agent, and is in a healthy state.

Bottom Line: `local` would be a special tag that when used would make the agent not consult the entirety of the datacenter and only resolve local services to local agent.

**Comment by highlyunavailable** [comment]
This is sort of an inverse duplicate of #488 that I asked for a long time ago and sort of related to #252 I think.

**Comment by slackpad** [comment]
https://github.com/hashicorp/consul/pull/2137 will let you do this using a simple prepared query definition (and with prepared query templates you can create a single template definition that'll cover all your services).
